classes/items/armor.cpp

```diff
@@ -12,8 +12,8 @@
 Armor::Armor(
         std::string name,
         int defense,
-        unsigned int tier,
         unsigned int type,
+        unsigned int tier,
         unsigned int value,
         unsigned int level)
        :Item(name, tier, type, value, level) //calls base constructor
```

Comment:
type shouldn't be set within the Armor constructor. You should instead go in the type slot so that an item that's constructed as armor couldn't be changed. further default parameters aren't set here so they won't be when doing the default construction. the compiler uses the definition not the declaration.

It'd be something like this(first remove the type argument).

Armor(ALL_ARGUMENTS)
:Item(name, tier, 2, value, level)